pkg/controller/kogitodataindex/resource/statefulset.go

```diff
@@ -127,3 +147,28 @@ func mountProtoBufConfigMaps(statefulset *appsv1.StatefulSet, cli *client.Client
 
 	return nil
 }
+
+// defineDataIndexHTTPPort will define which port the dataindex should be listening to. To set it use httpPort cr parameter.
+// defaults to 8080
+func defineDataIndexHTTPPort(instance *v1alpha1.KogitoDataIndex) int32 {
+	// first check if the env KOGITO_DATA_INDEX_HTTP_PORT is set
+	for env, value := range instance.Spec.Env {
```

Comment:
I think you don't need to do this. Just override the value for `KOGITO_DATA_INDEX_HTTP_PORT` in the container. Since we already have an input for `httpPort`in the CR, there's no need for the user to set this env. If they set, it's expected to be overrided.